stactools_landsat/stactools/landsat/utils.py

```diff
@@ -0,0 +1,141 @@
+    """
+    Handle USGS MTL as a dict and return a STAC item.
+
+    NOT IMPLEMENTED
+
+    Issues include:
+        - There's no reference to UTM Zone or any other CRS info in the MTL
+        - There's no absolute file path or reference to a URI to find data.
+    """
+    LANDSAT_METADATA = metadata["LANDSAT_METADATA_FILE"]
+    product = LANDSAT_METADATA["PRODUCT_CONTENTS"]
+    projection = LANDSAT_METADATA["PROJECTION_ATTRIBUTES"]
+    image = LANDSAT_METADATA["IMAGE_ATTRIBUTES"]
+    proessing_record = LANDSAT_METADATA["LEVEL2_PROCESSING_RECORD"]
+
+    scene_id = product["LANDSAT_PRODUCT_ID"]
+
+    xmin, xmax = float(projection["CORNER_LL_LON_PRODUCT"]), float(
+        projection["CORNER_UR_LON_PRODUCT"])
+    ymin, ymax = float(projection["CORNER_LL_LAT_PRODUCT"]), float(
+        projection["CORNER_UR_LAT_PRODUCT"])
+    geom = mapping(box(xmin, ymin, xmax, ymax))
+    bounds = shape(geom).bounds
+
+    # Like: "2020-01-01" for date and  "23:08:52.6773140Z" for time
+    dt = _parse_date(f"{image['DATE_ACQUIRED']}T{image['SCENE_CENTER_TIME']}")
+    created = _parse_date(proessing_record["DATE_PRODUCT_GENERATED"])
+
+    item = Item(id=scene_id,
+                geometry=geom,
+                bbox=bounds,
+                datetime=dt,
+                properties={})
+
+    # Common metadata
+    item.common_metadata.created = created
+    item.common_metadata.platform = image["SPACECRAFT_ID"]
+    item.common_metadata.instruments = [
+        i.lower() for i in image["SENSOR_ID"].split("_")
+    ]
+
+    # TODO: implement these three extensions
+    item.ext.enable("eo")
+    item.ext.enable("view")
+    item.ext.enable("projection")
+
+    return item
+
+
+def transform_stac_to_stac(item: Item,
+                           enable_proj: bool = True,
+                           self_link: str = None,
+                           source_link: str = None) -> Item:
+    """
+    Handle a 0.7.0 item and convert it to a 1.0.0.beta2 item.
+    """
+    # Remove USGS extension and add back eo
+    item.ext.enable("eo")
+
+    # Add and update links
+    item.links = []
+    if self_link:
+        item.links.append(Link(rel="self", target=self_link))
+    if source_link:
+        item.links.append(
+            Link(rel="derived_from",
+                 target=source_link,
+                 media_type="application/json"))
+
+    # Add some common fields
+    item.common_metadata.constellation = "Landsat"
+    item.common_metadata.instruments = [
+        i.lower() for i in item.properties["eo:instrument"].split("_")
+    ]
+    del item.properties["eo:instrument"]
+
+    # Handle view extension
+    item.ext.enable("view")
+    item.ext.view.off_nadir = item.properties["eo:off_nadir"]
+    del item.properties["eo:off_nadir"]
+
+    if enable_proj:
+        try:
+            # If we can load the blue band, use it to add proj information
+            blue_asset = item.assets["SR_B2.TIF"]
+            blue = rasterio.open(blue_asset.href)
```

Comment:
If you use the MTL file instead, I don't think you need to load the asset.. Specifically, the bounding box in the local crs is included, so you should be able to infer the transform

This should be all you need?
```json
{
  "MAP_PROJECTION": "UTM",
  "UTM_ZONE": "13",
  "GRID_CELL_SIZE_REFLECTIVE": "30.00",
  "REFLECTIVE_LINES": "8001",
  "REFLECTIVE_SAMPLES": "7871",
  "CORNER_UR_PROJECTION_X_PRODUCT": "378900.000",
  "CORNER_UR_PROJECTION_Y_PRODUCT": "4429200.000",
  "CORNER_LL_PROJECTION_X_PRODUCT": "142800.000",
  "CORNER_LL_PROJECTION_Y_PRODUCT": "4189200.000",
}
```

Reply:
MTL aside, I don't think there's CRS info in the STAC file either, which is a bit of an issue really.

Inferring CRS from the bounding box of the original STAC is fraught with danger, I think...